src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_TEXT = "supply";
+    private static final String BUY_TEXT = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String line = readFromFile(fromFileName);
+        String report = createReport(line);
+        writeToFile(toFileName, report);
+    }
+
+    public String readFromFile(String fromFileName) {
+        File incomingFile = new File(fromFileName);
+        String readString;
+        StringBuilder readerBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(incomingFile))) {
```

Comment:
ideally, we should keep the try section as short as possible, to isolate the potential error from the rest of the code

Reply:
will do, thx